demo04_basic_attacker.py

```diff
@@ -1,55 +1,59 @@
 # This bot selects a food pellet at random, then goes and tries to get it by
 # following the shortest path to it.
 # It tries on the way to avoid being eaten by the enemy: if the next move
-# to get to the food would put it on a ghost, then it steps back to its
-# previous position
+# to get to the food would put it on a ghost, then it chooses a random safe
+# position
 TEAM_NAME = 'Basic Attacker Bots'
 
 from pelita.utils import Graph
 
-from utils import next_step
+from utils import shortest_path
+
 
 def move(bot, state):
+    enemy = bot.enemy
     # we need to create a dictionary to keep information (state) along rounds
     # the state object will be passed untouched at every new round
     if state is None:
         # initialize the state dictionary
         state = {}
         # each bot needs its own state dictionary to keep track of the
         # food targets
-        state[0] = None
-        state[1] = None
+        state[0] = (None, None)
+        state[1] = (None, None)
         # initialize a graph representation of the maze
         # this can be shared among our bots
         state['graph'] = Graph(bot.position, bot.walls)
 
-    target = state[bot.turn]
+    target, path = state[bot.turn]
 
     # choose a target food pellet if we still don't have one or
     # if the old target has been already eaten
-    if (target is None) or (target not in bot.enemy[0].food):
-        target = state[bot.turn] = bot.random.choice(bot.enemy[0].food)
+    if (target is None) or (target not in enemy[0].food):
+        # position of the target food pellet
+        target = bot.random.choice(enemy[0].food)
+        # shortest path from here to the target
+        path = shortest_path(bot.position, target, state['graph'])
+        state[bot.turn] = (target, path)
```

Comment:
It’s a bit back-and-forth with the assigning to `state[bot.turn]` and `target, path` (sometimes inside an if, sometimes unconditionally). Maybe it would be better to have a single assignment at the end?